src/taskgraph/optimize/strategies.py

```diff
@@ -80,8 +80,8 @@ def check(self, files_changed, patterns):
         return False
 
     def should_remove_task(self, task, params, file_patterns):
-        # pushlog_id == -1 - this is the case when run from a cron.yml job or on a git repository
-        if params.get("repository_type") == "hg" and params.get("pushlog_id") == -1:
+        # skip-unless-changed should not apply to cron tasks (there will never be file changes)
+        if params.get("tasks_for") == "cron":
```

Comment:
I suspect the same is true for action tasks, as far as I can tell they'll also have identical base-rev and head-rev, unlike push and pull-request events.  Could we check for that instead of explicitly looking for cron?

Reply:
Right, I'm hoping that would do the trick (but it depends on what .taskcluster.yml does, in case projects differ from the template that might complicate things).